### PR TITLE
Add backend support for getting quests sorted by distance etc

### PR DESCRIPTION
**Pull request by magicus**
This is not used yet but will be in #367, and hopefully in the GUI as well :)

### CODE REVIEW

File: common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java [hunk 1]
```diff
@@ -300,11 +306,70 @@ public static List<QuestInfo> getQuests() {
...
+        return Optional.of(new Location(
+                Integer.parseInt(matcher.group(1)),
+                Integer.parseInt(matcher.group(2)),
+                Integer.parseInt(matcher.group(3))));
+    }
+
+    private static class LocationComparator implements Comparator<QuestInfo> {
+        private static final Vec3 PLAYER_LOCATION = McUtils.player().position();
+
+        private double getDistance(Optional<Location> loc) {
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java:345]: Is Qodana's suggestion relevant?

File: common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java [hunk 2]
```diff
@@ -300,11 +306,70 @@ public static List<QuestInfo> getQuests() {
...
+        @Override
+        public int compare(QuestInfo quest1, QuestInfo quest2) {
+            Optional<Location> loc1 = quest1.getNextLocation();
+            Optional<Location> loc2 = quest2.getNextLocation();
+            return (int) (getDistance(loc1) - getDistance(loc2));
+        }
+    }
+
+    public enum QuestSortOrder {
+        NONE,
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java:362]: UNORDERED?

**Review by kristofbolyai — COMMENTED** [review]
Looks good otherwise

File: common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java (hunk 2 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java:362]: Looking at this after a while, maybe NONE is better.

**Comment by kristofbolyai** [comment]
If this get's merged, I think I can add the GUI part tomorrow.

File: common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java (hunk 1 above)
Comment (magicus) [reply common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java:345]: Ah, found it:

>  'Optional' used as field or parameter type: common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java#L345
`Optional` used as type for parameter 'loc'

Comment (magicus) [reply common/src/main/java/com/wynntils/wynn/model/questbook/QuestBookManager.java:345]: (But not in the code review GUI, had to search in the GHA logs...)

No, I don't think that's relevant. I mean, it's kind of the point that we should take in an Optional and return 0 if it is empty. I get it why one should avoid Optional in fields, but I have a hard time understanding why they should not be passed as arguments. Seems quite valid to me.

**Review by kristofbolyai — APPROVED** [review]

**Pull request merged**